okio-wasifilesystem/src/wasmMain/kotlin/okio/WasiFileSystem.kt

```diff
@@ -427,13 +428,25 @@ object WasiFileSystem : FileSystem() {
 
   override fun createSymlink(source: Path, target: Path) {
     withScopedMemoryAllocator { allocator ->
+      val sourcePreopen = preopenForPath(source)
+        ?: throw FileNotFoundException("no preopen: $source")
+
+      // Always create symlinks relative to their source. Absolute symlinks are trouble because the
+      // absolute paths used by WASI are different from the absolute paths on the host file system.
+      val sourceParent = source.parent
+        ?: throw IOException("unexpected symlink source: $source")
+      val targetRelative = when {
+        target.isRelative -> target
+        else -> target.relativeTo(sourceParent)
+      }
```

Comment:
Is this even acknowledged as a bug by the WASI team?

Reply:
Yeah, they’ve changed it in the forthcoming preview2.